src/neurotechdevkit/results/_results.py

```diff
@@ -420,6 +424,182 @@ def render_steady_state_amplitudes_3d(
+        via `slice_axis` and `slice_position`. Eg. to take a slice at z=0.01 m, use
+        `slice_axis=2` and `slice_position=0.01`. If `slice_axis` and `slice_position`
+        are not specified, the slice will be taken along the scenario's `slice_axis` and
+        `slice_position`.
+
+        !!! warning
+            When taking a 2D slice of a 3D steady-state result, the sources and target
+            are not being accounted for and will not be preserved in the 2D result.
+
+        Args:
+            slice_axis (Optional[SliceAxis], optional): The axis along which to slice
+                the 3D field to be recorded. If None, then the complete field will be
+                recorded. Use 0 for X axis, 1 for Y axis and 2 for Z axis. Defaults to
+                None.
+            slice_position (Optional[float], optional): The position (in meters) along
+                the slice axis at which the slice of the 3D field should be made.
+                Defaults to None.
+
+        Returns:
+            A SteadyStateResult2D object containing the 2D slice of the 3D steady-state
+            simulation results.
+        """
+        if slice_axis is None and slice_position is None:
+            slice_axis = self.scenario.slice_axis
+            slice_position = self.scenario.slice_position
+
+        self.scenario._validate_slice_args(slice_axis, slice_position)
+
+        assert slice_axis is not None
+        assert slice_position is not None
+
+        sources_2d = []
+        source_type = {
+            sources.FocusedSource3D: sources.FocusedSource2D,
+            sources.PlanarSource3D: sources.PlanarSource2D,
+            sources.PhasedArraySource3D: sources.PhasedArraySource2D,
+        }
+        for source in self.scenario.sources:
+            assert type(source) in source_type
+            source_args = {
+                "position": drop_element(source.position, slice_axis),
+                "direction": drop_element(source.unit_direction, slice_axis),
+                "num_points": source.num_points,
+            }
+            if isinstance(source, sources.PhasedArraySource3D):
+                if source.element_delays is None:
+                    source_args["tilt_angle"] = source.tilt_angle
+                    source_args["focal_length"] = source.focal_length
+                    source_args["delay"] = source.delay
+                source_args["num_elements"] = source.num_elements
+                source_args["pitch"] = source.pitch
+                source_args["element_width"] = source.element_width
+                source_args["element_delays"] = source.element_delays
+            else:
+                source_args["aperture"] = source.aperture
+                source_args["delay"] = source.delay
+                if not isinstance(source, sources.UnfocusedSource):
+                    source_args["focal_length"] = source.focal_length
+            sources_2d.append(source_type[type(source)](**source_args))
```

Comment:
thanks for taking this on! I didn't think about converting the `source`s and `target`s when writing the ticket; it does add a decent amount of complexity.